src/nvim/ops.c

```diff
@@ -3398,7 +3398,7 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
         bcount_t totsize = 0;
         int lastsize = 0;
         if (y_type == kMTCharWise
-            || (y_type == kMTLineWise && flags & PUT_LINE_SPLIT)) {
```

Comment:
hmm this looks messy and weird, and this if condition is not correct (need to duplicate the new check, not remove this one).

Can't the issue be solved just by adjusting the if-statement around `mark_adjust` below, which seems like the real root case? "Skip mark_adjust when adding lines after the last one"

Reply:
to be clear this is not just for buf_updates, there _can_ be an right-gravity ext-mark past the end of the buffer.

Reply:
The thing is that we lose the total put length in the mark_adjsut below, so that's my issue

Reply:
lose how? mark_adjust always calculates the length, this is not a special case.

Reply:
Oh you mean do the extmark move in mark_adjust ?

Reply:
Yes, just try loosen the condition, i see no reason it wouldn't work.

Reply:
eheh, it indeed works

Reply:
I think I fixed the issue, could you confirm the fix suits you ?

Reply:
Lgtm